hydra-cluster/test/Test/HydraExplorerSpec.hs

```diff
@@ -0,0 +1,239 @@
+
+                headExplorerSees explorer "HeadCommitTx" headId
+                headExplorerSees explorer "HeadCollectComTx" headId
+
+                send hydraNode $ input "Close" []
+
+                headExplorerSees explorer "HeadCloseTx" headId
+
+                waitFor hydraTracer 50 [hydraNode] $
+                  output "ReadyToFanout" ["headId" .= headId]
+
+                send hydraNode $ input "Fanout" []
+
+                headExplorerSees explorer "HeadFanoutTx" headId
+
+  it "can observe hydra transactions created by multiple hydra-nodes" $
+    failAfter 60 $
+      showLogsOnFailure "HydraExplorerSpec" $ \tracer -> do
+        withTempDir "hydra-explorer-history" $ \tmpDir -> do
+          withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \cardanoNode@RunningNode{nodeSocket} -> do
+            let hydraTracer = contramap FromHydraNode tracer
+            hydraScriptsTxId <- publishHydraScriptsAs cardanoNode Faucet
+
+            let initHead hydraNode = do
+                  send hydraNode $ input "Init" []
+                  waitMatch 5 hydraNode $ \v -> do
+                    guard $ v ^? key "tag" == Just "HeadIsInitializing"
+                    v ^? key "headId" . _String
+
+            (aliceCardanoVk, _aliceCardanoSk) <- keysFor Alice
+            seedFromFaucet_ cardanoNode aliceCardanoVk 100_000_000 (contramap FromFaucet tracer)
+            aliceChainConfig <- chainConfigFor Alice tmpDir nodeSocket hydraScriptsTxId [] cperiod
+            aliceHeadId <- withHydraNode hydraTracer aliceChainConfig tmpDir 1 aliceSk [] [1] initHead
+
+            (bobCardanoVk, _bobCardanoSk) <- keysFor Bob
+            seedFromFaucet_ cardanoNode bobCardanoVk 100_000_000 (contramap FromFaucet tracer)
+            bobChainConfig <- chainConfigFor Bob tmpDir nodeSocket hydraScriptsTxId [] cperiod
+            bobHeadId <- withHydraNode hydraTracer bobChainConfig tmpDir 2 bobSk [] [2] initHead
+
+            withHydraExplorer cardanoNode $ \explorer -> do
+              headExplorerSees explorer "HeadInitTx" aliceHeadId
+              headExplorerSees explorer "HeadInitTx" bobHeadId
+
+  it "can query for all hydra heads observed" $
+    failAfter 60 $
+      showLogsOnFailure "HydraExplorerSpec" $ \tracer -> do
+        withTempDir "hydra-explorer-get-heads" $ \tmpDir -> do
+          withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \cardanoNode@RunningNode{nodeSocket} -> do
+            let hydraTracer = contramap FromHydraNode tracer
+            hydraScriptsTxId <- publishHydraScriptsAs cardanoNode Faucet
+            withHydraExplorer cardanoNode $ \explorer -> do
+              (aliceCardanoVk, _aliceCardanoSk) <- keysFor Alice
+              aliceChainConfig <- chainConfigFor Alice tmpDir nodeSocket hydraScriptsTxId [] cperiod
+              aliceHeadId <- withHydraNode hydraTracer aliceChainConfig tmpDir 1 aliceSk [] [1] $ \hydraNode -> do
```

Comment:
No big deal but maybe we want to set the hydra keys here also (second to last argument) to `withHydraNode`. This is also true for the test above.

Reply:
Actually not needed since you run just a single party heads here.